internal/core/block/block.go

```diff
@@ -82,6 +82,11 @@ type DAGLink struct {
 	// Name is the name of the link.
 	//
 	// This will be either the field name of the CRDT delta or "_head" for the head link.
+	//
+	// This field currently serves no purpose and is duplicating data already held on the target
+	// block.  However we want to have this long term to enable some fancy P2P magic to allow users
+	// to configure the collection to only sync particular fields using
+	// [GraphSync](https://github.com/ipfs/go-graphsync) which will need to make use of this property.
```

Comment:
suggestion: You could just say that this field is needed for graphSync and sync optimisation instead of (not sure this was you intention but this is what I get from this) showing through the comment that you are annoyed that the field is still there.

Reply:
No intended annoyance in the wording, I do feel that it is important to note that the current lack of purpose is known and somewhat deliberate though otherwise people (including future me) will assume the lack of purpose is accidental and try and remove it,